src/sql/workbench/contrib/objectExplorer/browser/serverTreeView.ts

```diff
@@ -663,9 +638,56 @@ export class ServerTreeView extends Disposable implements IServerTreeView {
 				},
 				getActionsContext: () => (actionContext)
 			});
+		}
+	}
 
-			return true;
+	private async onTreeNodeDoubleClick(node: ServerTreeElement): Promise<void> {
+		const action = this._actionProvider.getDefaultAction(this.tree, node);
+
+		if (action) {
+			this._actionRunner.run(action, this.getActionContext(node)).catch(errors.onUnexpectedError);
+		} else {
+			// If no default action is defined, fallback to the default behavior of opening the dashboard.
+			// Open dashboard on double click for server and database nodes
```

Comment:
should we make opening the dashboard the default command for server/db nodes, at least for MSSQL?  At least that way we could "deprecate" this fallback.

Reply:
I thought about it but didn't implement it that way because it will make us not able to contribute default action for these 2 node types from extensions. we can do that if in the future we allow multiple default actions and use priority/order to decide which one wins. but for now I think it is ok to leave it like this since the code duplication isn't that bad.